draft-aegis-mimi-arch-latest.md

```diff
@@ -43,28 +43,166 @@ TODO Abstract
 
 # Introduction
 
-## Scope:
-* interface between AS and MLS -- something AS has to implement for both
-  client and back-end
-* interface between AS and application, split into 3 components
-* Overview of services and very high-level of how they are used together.
-* ?? how components federate
+In order to achieve cross-application federation between a set of MLS applications there must be a common set of APIs available that implement the basic AS/DS requirements as defined by `MLS Architecture`. To facilitate the creation of groups asynchronously, MLS clients may have a need to establish identity, credentials and a queue of key packages. Once groups are established, it is useful for clients to have access to a common repository for welcome messages, pending proposals, commit messages, and ratchet trees. 
 
-## Actors
-* back-end
-* application
-* client
+In this document we describe a MIMI Gateway API that can be used to provide intra domain and cross domain federation between MLS applications.

```

Comment:
I'm missing how identity provider fits into the bigger picture? Currently it appears out of the blue in the next section. So I would

1. make this more general, like "we describe the API of AS and DS components that can be used to provide intra domain and cross domain federation between MLS applications. These components include  MIMI Gateway API and Identity Provider".
2. in the operational context below, place the provider in the bigger picture, something like "To ensure consistent behavior, all actors that execute MLS, including all clients and Gateways, must agree on its AS components (e.g., credential types used by MLS). The AS-related functionality that actors need to agree on is collected in the Identity Provider"

Reply:
Ran out of time today, can you try to address that?